azkaban-common/src/main/java/azkaban/project/Project.java

```diff
@@ -39,6 +39,7 @@
   private int version = -1;
   private long createTimestamp;
   private long lastModifiedTimestamp;
+  private String currentUploaderID;
```

Comment:
Yes, it is the ID of the latest uploader who tried uploading to azkaban be it successful/not. It is transient as there is no DB update. There exists a field lastModifiedUser which gets persisted after the upload succeeds and is fetched from the DB.

What do you mean by all code paths? Why do you think it is optional?